### PR TITLE
feat(openapi-sync): preserve user-configured request values on sync

### PR DESCRIPTION
**Pull request by sundram-bruno**
### Description

Discussion: #7401

On the **Spec Updates** sync, Bruno currently overwrites a request's body, params, headers and auth wholesale, destroying user-entered values and `{{envVar}}` references whenever the spec changes shape. This PR makes sync reconcile the request *structure* against the spec while *preserving the user's values* for fields that still exist.

JIRA : https://usebruno.atlassian.net/browse/BRU-3205

A **"Preserve values"** toggle (default ON) on the Spec Updates review toolbar controls this. With it on, your edits are kept and only the spec's structural additions/removals are applied; with it off, spec values overwrite (the seam for an explicit "take the spec's version"). The diff preview's EXPECTED column shows the post-merge result, so unchanged values no longer render as changes.

**What changed**

Backend — `packages/bruno-electron/src/ipc/openapi-sync.js`
- Field-level merge: JSON body merged by key path; form fields and params/headers merged by name (duplicate names paired positionally), preserving `value` + `enabled`.
- `{{var}}` masking so an interpolated body can be parsed, merged and restored safely — using a private-use sentinel that never collides with body text and never reaches disk.
- Auth merged by mode: same mode keeps the user's values and adds spec-introduced fields; a mode change takes the spec. `compareRequestFields` now compares auth by mode only, so preserved auth values no longer keep the collection permanently "out of sync".
- `preserveValues` (default `true`) threaded through `renderer:apply-openapi-sync` and `renderer:get-endpoint-diff-data`. Reset path left unchanged; scripts/tests/assertions preserved in both sync and reset.

Frontend — `OpenAPISyncTab`
- "Preserve values" toggle on the Spec Updates review; per-endpoint preview re-fetches when toggled; the flag is forwarded to apply.

image

**Note for reviewers — one intended behavior change:** `compareRequestFields` no longer flags auth *config* differences (e.g. apikey placement, oauth2 url/scope) when the auth mode is unchanged; only a mode change is drift. This is required so preserved auth values don't perpetually re-flag the endpoint.

**Tests:** 67 unit tests in `packages/bruno-electron/tests/ipc/openapi-sync-merge.spec.js` covering the merge helpers and `{{var}}` masking edge cases (string boundaries, object keys, nested/array bodies, many vars, dual-side). Full `bruno-electron` and `bruno-app` suites pass.

#### Contribution Checklist:

- [x] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes** — see the auth drift-detection note above (behavior change, not an API break).
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.** — #7548

Note: Keeping the PR small and focused helps make it easier to review and merge.

## Summary by CodeRabbit

* **New Features**
  * "Preserve values" toggle in OpenAPI sync review and row-level controls; applies to bulk apply actions and preserves user-edited request data when enabled (default on, accessible with tooltip).
* **Bug Fixes**
  * More reliable diff loading and UI: stale async results ignored, spinner/error shown only on initial loads to avoid flicker during refetches.
* **Tests**
  * Comprehensive tests validating merge/preserve behaviors across bodies, fields, auth, and sync/apply flows.
* **Style**
  * New compact "preserve values" chip and switch styling.

### CODE REVIEW

File: packages/bruno-app/src/components/OpenAPISyncTab/StyledWrapper.js [hunk 1]
```diff
@@ -1804,6 +1804,61 @@ const StyledWrapper = styled.div`
...
+        height: 14px;
+        border-radius: 7px;
+        border: none;
+        padding: 0;
+        flex-shrink: 0;
+        position: relative;
+        cursor: pointer;
+        transition: background 0.2s;
+        background: ${(props) => props.theme.colors.text.muted};
+
```
[the remark this answers is not in the document]
Comment (sundram-bruno) [reply packages/bruno-app/src/components/OpenAPISyncTab/StyledWrapper.js:1851]: added changes.

**Review by sachin-bruno — APPROVED** [review]

**Review by bijin-bruno — APPROVED** [review]
